packages/typescript-estree/src/node-utils.ts

```diff
@@ -766,3 +766,18 @@ export function getContainingFunction(
 ): ts.SignatureDeclaration | undefined {
   return ts.findAncestor(node.parent, ts.isFunctionLike);
 }
+
+// `ts.nodeCanBeDecorated`
+export function nodeCanBeDecorated(node: ts.Node): boolean {
+  return [true, false].some(
+    useLegacyDecorators =>
+      // @ts-expect-error -- internal api
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+      ts.nodeCanBeDecorated?.(
```

Comment:
instead of making this reference - can we add it to the TS types [in our ts augmentation](https://github.com/typescript-eslint/typescript-eslint/blob/3db653276b97d79f9b0c93b48f4fd0b62d58f694/packages/typescript-estree/typings/typescript.d.ts)